src/Compilers/Core/RebuildTest/OptionRoundTripTests.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System;
+using System.Reflection.PortableExecutable;
+using BuildValidator;
+using Castle.Core.Logging;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Emit;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.Extensions.Logging;
+using Xunit;
+using System.IO;
+using System.Threading;
+using Microsoft.CodeAnalysis.VisualBasic;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.Rebuild.UnitTests
+{
+    public class OptionRoundTripTests : CSharpTestBase
+    {
+        public static readonly object[][] s_platforms = ((Platform[])Enum.GetValues(typeof(Platform))).Select(p => new[] { (object)p }).ToArray();
+
+        [Theory]
+        [MemberData(nameof(s_platforms))]
+        public void Platform_RoundTrip(Platform platform)
+        {
+            const string path = "test";
+
+            var original = CreateCompilation(
+                "class C { static void Main() { } }",
+                options: TestOptions.DebugExe.WithPlatform(platform));
+
+            original.VerifyDiagnostics();
+
+            var originalBytes = original.EmitToArray(new EmitOptions(debugInformationFormat: DebugInformationFormat.Embedded));
+            var peReader = new PEReader(originalBytes);
+            Assert.True(peReader.TryOpenAssociatedPortablePdb(path, path => null, out var provider, out _));
+            var pdbReader = provider!.GetMetadataReader();
+
+            var factory = LoggerFactory.Create(configure => { });
+            var logger = factory.CreateLogger(path);
+            // TODO: shouldn't need to pass a logger.
+            var bc = new BuildConstructor(logger);
+
+            var optionsReader = new CompilationOptionsReader(logger, pdbReader, peReader);
+
+            var sources = original.SyntaxTrees.Select(st =>
+            {
+                var text = st.GetText();
+                return new ResolvedSource(OnDiskPath: null, text, new SourceFileInfo(path, text.ChecksumAlgorithm, text.GetChecksum().ToArray(), text, embeddedCompressedHash: null));
+            }).ToImmutableArray();
```

Comment:
>ToImmutableArray [](start = 15, length = 16)

Could use `SyntaxTrees.SelectAsArray(...)` instead.

Reply:
I think this project is missing an IVT to MS.CA for this. I will punt on it for now if that's ok.